src/main/java/sigma/Parser.java

```diff
@@ -0,0 +1,188 @@
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import sigma.task.*;
+
+/**
+ * Class that handles the logic relating to user commands
+ */
+public class Parser {
+    private FileWriter writer;
+
+    /**
+     * Constructor for a Parser
+     */
+    public Parser() {
+        try {
+            writer = new FileWriter("data/sigma.txt", true);
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+
+    /**
+     * Greets the user upon command
+     * @return The greeting message
+     */
+    public String greet() {
+        return "hello, I'm Sigma, your personal chatbot";
+    }
+
+    /**
+     * Greets the user goodbye and closes the program
+     */
+    public void goodbye() {
+        try {
+            writer.flush();
+            writer.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        } finally {
+            System.out.println("leaving so soon? goodbye!");
+        }
+    }
+
+    /**
+     * Adds a To-do task to the TaskList
+     * @param userInput The to-do task that the user wants to add
+     */
+    public void handleTodo(String userInput) {
+        String description = userInput.substring(4).trim(); // trims the input away
+        if (description.isEmpty()) { // handle empty input
+            System.out.println("todo...todo what? let's try this again");
+            return;
+        }
+        Task task = new ToDo(description, false);
+        TaskList.addToList(task);
+        try {
+            writer.write(task + "\n");
+        } catch (IOException exception) {
+            System.err.println("an error occurred writing to file: " + exception.getMessage());
+        }
+        System.out.println("added todo task:\n [T][ ] " + description);
+    }
+
+    /**
+     * Adds a Deadline task to the TaskList
+     * @param userInput The Deadline task that the user wants to add
+     */
+    public void handleDeadline(String userInput) {
+        Pattern pattern = Pattern.compile("deadline (.+) /by (.+)");
+        Matcher matcher = pattern.matcher(userInput);
+        if (matcher.find()) {
+            String description = matcher.group(1);
+            String by = matcher.group(2);
+            String dateBy = stringToDate(by);
+            Task task = new Deadline(description, false, dateBy);
+            TaskList.addToList(task);
+            try {
+                writer.write(task + "\n");
+            } catch (IOException exception) {
+                System.err.println("an error occurred writing to file");
+            }
+            System.out.println("added deadline task:\n  [D][ ] " + description + " (by: " + dateBy + ")");
+        } else {
+            System.out.println("is the deadline in the room with us? let's try again");
+        }
+    }
+
+    /**
+     * Adds an Event task to the TaskList
+     * @param userInput The Event task that the user wants to add
```

Comment:
As mentioned by @tckeong, just to remind you to fix the JavaDoc in this file also 😺